### ISSUE TITLE
  Remember last session per user

### ISSUE THREAD

**Issue by martin21** [sddm/sddm#510]
kdm – Plasma´s former standard display manager – remembers the last session per user so it always chooses the correct last session for every user in case you have use more than one user on one system like I do with my laptop to separate company related and private stuff.

sddm according to `/var/lib/sddm/state.conf` only seems to remember one last session:

```
# Name of the session file of the last session selected. This session will be preselected when the login screen shows up.
Session=/usr/share/xsessions/plasma.desktop
```

For me it doesn´t matter all that much, cause I use the same session type for both users, but on real multiseat systems with users with different desktop environment preferences this is limiting.

I am reporting this here, since I did not find a sddm product on bugs.kde.org.

Issue #509 Remember last username per tty is related.

**Comment by davidedmundson** [comment]
Another argument for doing an AccountsService port. 
We'd get this for free as the xsessions property of the user would store this. (though only once, not per TTY)

**Comment by jleclanche** [comment]
@davidedmundson  :+1:

**Comment by rdieter** [comment]
dup of issue #481

**Comment by kkofler** [comment]
Both this and issue #481 are really duplicates of issue #346.
(See also the related issue #326, which is about a system-wide default session.)

**Comment by tpgxyz** [comment]
Ok this means that with Calamres installer it will be not possible to set default session to log in.
Without this users will get confused as they will log into failsafe session rather that Plasma5 because on sddm login screen a "Failsafe" session was set up as default.

**Comment by tmccombs** [comment]
Any progress on this?

**Comment by debalance** [comment]
Please fix this!

**Comment by thalunil** [comment]
I would also appreciate if sddm remembers the last session of an user...
Nobody is interested?

**Comment by tghgg** [comment]
Really hoping for this to be resolved. Please, devs!

**Comment by RJVB** [comment]
Having made a tour of the currently available display managers/greeters I can say that there is basically none that suits and pleases me as well as the late KDM.

KDM was part of part of the kde-workspace bundle but it should be possible to 1) isolate it and 2) port it to Qt5/KF5 . After that it should just be another display manager and with a bit of luck all the old themes will continue to work (although I use it without any themes, with just the nice analog clock in the old-style greeter and a few handpicked nice fonts).

Anyone else interested?

**Comment by martin21** [comment]
I am not into porting KDM at the moment. I'd like something that is supported. I consider switching to LigthtDM. However discussing things like this is off topic here.

**Comment by RJVB** [comment]
I didn't propose to discuss that here.

**Comment by darkshvein** [comment]
sddm Stilll did not remember last session.
Hello from 2025 year.

**Comment by Vogtinator** [comment]
PRs welcome...

Could also be implemented in a simpler way by just saving a per-user default in the state file (e.g. `Session[username]`). The tricky part is again theme integration: It has to change the selected session after changing the user name.

**Comment by groszdaniel** [comment]
> Could also be implemented in a simpler way by just saving a per-user default in the state file (e.g. `Session[username]`). The tricky part is again theme integration: It has to change the selected session after changing the user name.

If it doesn't change after the username is entered, but rather just shows something like "Default/Last used" unless you manually choose one, that's still better than nothing. Also, preferably store it in the user's home directory (`~/.dmrc`) so that it's remembered per user across machines on a network with shared users and home directories.

**Comment by MacsKavana** [comment]
> ented in a simpler way by just saving a per-user default in the state file (e.g. `Session[username]`). The tricky part is again theme integration: It has t

I think people is less interested in any kind of theme integration than to have their favourite DE immediately after entering their password. It's really very annoying to always switch DE during login. This could be done by writing a simple file somewhere in the [XDG user dirs](https://www.freedesktop.org/wiki/Software/xdg-user-dirs/) directories.

**Comment by RJVB** [comment]
> I think people is less interested in any kind of theme integration than to have their favourite DE 

Exactly - what theme would even have to be switched?


I'm reminded that I spent some time porting KDM to Qt5 but still haven't bothered testing it ^^

**Comment by christianlegge** [comment]
I ended up working around this with a hack script. As long as you know your users (e.g. family sharing a physical machine), it should work fine. It gives more overhead to switching between DEs as a single user as well, if that bothers you. I'm sure there are many ways around that if you need.

`/usr/local/bin/launch_preferred_session`
```bash
#!/usr/bin/env bash

if [ $USER == "myuser" ]; then
    # can be any session you want, of course
    /usr/lib/plasma-dbus-run-session-if-needed /usr/bin/startplasma-wayland
else # with more if clauses if desired...
    Hyprland
fi
```

Then create `/usr/share/wayland-sessions/preferred.desktop` and set its `Exec` to the above script. Choose it in sddm and never touch it again.
